compiler/grainformat/reformat.re

```diff
@@ -3308,258 +4127,177 @@ let print_primitive_value_description =
+        ~comments,
+        value_bindings,
+      )
+    | PTopExpr(expression) =>
+      print_expression(~original_source, ~comments, expression)
+    | PTopException(export_flag, type_exception) =>
+      let export =
+        switch (export_flag) {
+        | Nonexported => Doc.nil
+        | Exported => Doc.text("export ")
+        };
+      let cstr = type_exception.ptyexn_constructor;
+
+      let kind =
+        switch (cstr.pext_kind) {
+        | PExtDecl(sargs) =>
+          switch (sargs) {
+          | PConstrSingleton => Doc.nil
+          | PConstrTuple(parsed_types) =>
+            if (List.length(parsed_types) > 0) {
+              Doc.concat([
+                Doc.lparen,
+                Doc.join(
+                  Doc.comma,
+                  List.map(
+                    t =>
+                      print_type(
+                        ~original_source,
+                        ~comments,
+                        ~trailing_separator=false,
+                        t,
+                      ),
+                    parsed_types,
                   ),
-                  Doc.rparen,
-                ]);
-              } else {
-                Doc.nil;
-              }
+                ),
+                Doc.rparen,
+              ]);
+            } else {
+              Doc.nil;
             }
+          }
 
-          | PExtRebind(lid) => print_ident(lid.txt)
-          };
+        | PExtRebind(lid) => print_ident(lid.txt)
+        };
 
-        Doc.concat([
-          export,
-          Doc.text("exception "),
-          Doc.text(cstr.pext_name.txt),
-          kind,
-        ]);
-      | PTopExport(export_declarations) =>
-        Doc.concat([
-          Doc.text("export "),
-          Doc.join(
-            Doc.concat([Doc.comma, Doc.line]),
-            List.map(e => print_export_declaration(e), export_declarations),
-          ),
-        ])
+      Doc.concat([
+        export,
+        Doc.text("exception "),
+        Doc.text(cstr.pext_name.txt),
+        kind,
+      ]);
+    | PTopExport(export_declarations) =>
+      Doc.concat([
+        Doc.text("export "),
+        Doc.join(
+          Doc.concat([Doc.comma, Doc.line]),
+          List.map(print_export_declaration, export_declarations),
+        ),
+      ])
 
-      | PTopExportAll(export_excepts) =>
-        Doc.concat([
-          Doc.text("export * "),
-          if (List.length(export_excepts) > 0) {
-            Doc.concat([
-              Doc.text("except "),
-              Doc.join(
-                Doc.comma,
-                List.map(
-                  (excpt: Parsetree.export_except) =>
-                    switch (excpt) {
-                    | ExportExceptData(data) => Doc.text(data.txt)
-                    | ExportExceptValue(value) => Doc.text(value.txt)
-                    },
-                  export_excepts,
-                ),
+    | PTopExportAll(export_excepts) =>
+      Doc.concat([
+        Doc.text("export * "),
```

Comment:
is there always a space after `*`? Seems wrong

Reply:
Moved to the correct place